### PR TITLE
Add user route rebranched

### PR DESCRIPTION
**Pull request by stephenjmcneill1971**
Changes applied for ticket

https://dsdmoj.atlassian.net/jira/software/c/projects/CDS2/boards/994?modal=detail&selectedIssue=CDS2-125

To help the team understand how the journey should work in various contexts, the team have defined a comprehensive list of [user roles](https://docs.google.com/document/d/1NAy8lgMiWX9rRsr176VKSIHO_GLTgx75az2nsL0gTMQ/edit). The system currently supports showing personalised content based on these roles, however this functionality does not extend to page routing. To route specific roles to an appropriate page, we currently duplicate the rules for a role and use them directly in the routing definition.

To avoid this duplication and to help make the flow easier to comprehend, we are looking to re-use the roles functionality for routing purposes.

### CODE REVIEW

File: src/lib/resource/sections/download-your-answers.js [hunk 1]
```diff
@@ -37,30 +37,7 @@ module.exports = {
...
-                                    false
-                                ],
-                                [
-                                    '==',
-                                    '$.answers.p-applicant-are-you-18-or-over.q-applicant-are-you-18-or-over',
-                                    true
-                                ]
-                            ]
-                        ]
+                        ['or', ['|role.all', 'childUnder12'], ['|role.all', 'adult', 'incapable']]
```
[the remark this answers is not in the document]
Comment (neil-stephen-mcgonigle) [reply src/lib/resource/sections/download-your-answers.js:40]: removing the need for ['==', '$.answers.p--has-legal-authority.q--has-legal-authority', false]

File: src/lib/resource/sections/download-your-answers.js [hunk 2]
```diff
@@ -76,14 +53,7 @@ module.exports = {
...
                         ],
-                        [
-                            'dateCompare',
-                            '$.answers.p-applicant-enter-your-date-of-birth.q-applicant-enter-your-date-of-birth', // this date ...
-                            '>=', // is greater than or equeal too ...
-                            '-12', // 12 ...
-                            'years' // years (before, due to the negative (-12) ...
-                            // today's date (no second date given. defaults to today's date).
-                        ]
+                        ['|role.all', 'childOver12']
```
Comment (neil-stephen-mcgonigle) [inline src/lib/resource/sections/download-your-answers.js:56]: ['|role.all', 'childOver12', 'rep']

File: src/lib/resource/sections/applicant-unable-to-work.js
Comment (neil-stephen-mcgonigle) [inline src/lib/resource/sections/applicant-unable-to-work.js:1]: target: 'p-applicant-unable-to-work-duration',
                    cond: [
                        '==',
                        '$.answers.p-applicant-unable-to-work.q-applicant-unable-to-work',
                        true
                    ]
                },
                {
                    target: 'p-applicant-future-work',
                    cond: [
                                        ['|role.all', 'proxy', 'child'],
                    ]
                },
                {
                    target: 'p-applicant-affect-on-daily-life-dmi'
                }

Comment (neil-stephen-mcgonigle) [reply src/lib/resource/sections/applicant-unable-to-work.js:1]: image
tests are interesting always NO for mainapplicant and YES for reps?

File: src/lib/resource/sections/flow-has-legal-authority.js [hunk 3]
```diff
@@ -112,16 +112,10 @@ module.exports = {
...
-                            '$.answers.p-applicant-enter-your-date-of-birth.q-applicant-enter-your-date-of-birth', // this date ...
-                            '<', // is greater than or equal to ...
-                            '-18', // 18 ...
-                            'years' // years (before, due to the negative (-18) ...
-                            // today's date (no second date given. defaults to today's date).
-                        ]                           
+                    cond: [
+                        'and',
+                        ['|role.all', 'child'],
+                        ['==', '$.answers.p--has-legal-authority.q--has-legal-authority', false],
```
[the remark this answers is not in the document]
Comment (stephenjmcneill1971) [reply src/lib/resource/sections/flow-has-legal-authority.js:118]: Changed as requested all relevant UI tests still pass

File: src/lib/resource/sections/flow-has-legal-authority.js [hunk 4]
```diff
@@ -130,16 +124,7 @@ module.exports = {
...
-                            '==',
-                            '$.answers.p-applicant-are-you-18-or-over.q-applicant-are-you-18-or-over',
-                            true
-                        ],
-                        [
-                            '==',
-                            '$.answers.p-applicant-can-handle-affairs.q-applicant-capable',
-                            false
-                        ]
+                        ['|role.all', 'adult', 'incapable']
```
[the remark this answers is not in the document]
Comment (stephenjmcneill1971) [reply src/lib/resource/sections/flow-has-legal-authority.js:127]: Changed as requested all relevant UI tests still pass

File: src/lib/resource/sections/download-your-answers.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (stephenjmcneill1971) [reply src/lib/resource/sections/download-your-answers.js:40]: Changed as requested all relevant UI tests still pass

File: src/lib/resource/sections/download-your-answers.js (hunk 2 above)
Comment (stephenjmcneill1971) [reply src/lib/resource/sections/download-your-answers.js:56]: Changed as requested all relevant UI tests still pass

Comment (stephenjmcneill1971) [reply src/lib/resource/sections/download-your-answers.js:56]: Changed as requested all relevant UI tests still pass

File: src/lib/resource/sections/applicant-unable-to-work.js
Comment (stephenjmcneill1971) [reply src/lib/resource/sections/applicant-unable-to-work.js:1]: Only partially implemented change. Removed the extra p-applicant-affect-on-daily-life-dmi target section. The other changes caused UI test failures.

**Pull request merged**